pysindy/optimizers/stlsq.py

```diff
@@ -206,7 +206,8 @@ def _reduce(self, x, y):
         n_samples, n_features = x.shape
         n_targets = y.shape[1]
         n_features_selected = np.sum(ind)
-        sparse_sub = np.array(self.sparse_ind)
+        sparse_fix = [np.array(self.sparse_ind)] * y.shape[1]
```

Comment:
I had to create 2 similar list of arrays from sparse_ind, one of them remains fixed and the other changes according to the remove_and_decrement function.